test/index.js

```diff
@@ -163,4 +163,38 @@ test('concat', t => {
   t.done()
 })
 
+test('proxy', t => {
+  const testOpts = puddin({
+    a: {},
+    b: {}
+  })
+  let opts = testOpts({a: 1})
+  t.equal('a' in opts, true, 'provided opt detected')
+  t.equal('b' in opts, false, 'missing opt undetected')
+  t.equal('c' in opts, false, 'undefined opt undetected')
+  t.equal(opts.a, 1, 'provided key retrieved')
+  t.doesNotThrow(
+    () => {
+      opts.__woo = 'woo'
```

Comment:
I wasn't sure if this sort of thing was actually desirable, but the coverage completist in me could not resist.